tools/test/test-setup.sh

```diff
@@ -63,7 +63,7 @@ is_absolute "$TEST_UNDECLARED_OUTPUTS_ANNOTATIONS_DIR" ||
 
 is_absolute "$TEST_SRCDIR" || TEST_SRCDIR="$PWD/$TEST_SRCDIR"
 is_absolute "$TEST_TMPDIR" || TEST_TMPDIR="$PWD/$TEST_TMPDIR"
-is_absolute "$HOME" || HOME="$PWD/$TEST_TMPDIR"
+is_absolute "$HOME" || HOME="$PWD/$HOME"
```

Comment:
The test encyclopedia says that HOME should be set to TEST_TMPDIR:
https://docs.bazel.build/versions/master/test-encyclopedia.html

There's still a bug in the original code, which is that TEST_TMPDIR is already absolute, so this would double the $PWD prefix if HOME is not absolute (or not set?).

Setting HOME to TEST_TMPDIR seems to make sense to me for hermeticity reasons, unless the user explicitly requests it to be forwarded from outside the build, or sets it to an absolute path. Who or what is currently setting HOME for tests? It's not StandaloneTestStrategy, maybe it's always set by the shell?

Reply:
I agree setting HOME to TEST_TMPDIR makes sense.

On Linux, Bazel doesn't set HOME.
On Windows, MSYS sets it to the user's home directory's absolute path.
The else-branch is only used if the user passes `--test_env=HOME=foo` and then indeed `PWD` is duplicated.